aptos-move/mvhashmap/src/versioned_group_data.rs

```diff
@@ -361,9 +360,9 @@ impl<
         group_key: &K,
         txn_idx: TxnIndex,
         tags: HashSet<&T>,
-    ) -> Result<BTreeSet<(TxnIndex, Incarnation)>, PanicError> {
-        let mut invalidated_dependencies = BTreeSet::new();
-        self.remove_impl::<true>(group_key, txn_idx, tags, &mut invalidated_dependencies)?;
+    ) -> Result<BTreeMap<TxnIndex, Incarnation>, PanicError> {
+        let invalidated_dependencies = RegisteredReadDependencies::new();
```

Comment:
So every remove / write also creates a Mutex which is later dropped via `take_dependencies`? Should we have `Mutex<RegisteredReadDependencies>`?

Reply:
great point, added the usecase later and didn't notice - done.